Modifications for 5.0.0

TBG-FR opened the pull request:
This PR aims to solve (almost) all issues found in RocketChat `5.0.0`. As always, some (generated, and regular) classes changed, breaking some elements.

- ✅ Main chat messages (separator, action menu, popover ...)
- ✅ Thread chat messages 
- ✅ Admin panel information
- ✅ Admin panel sections
- ✅ Users information (Admin)
- ✅ Channel/Discussion tags (without breaking user roles tags, thanks @KeysBytes 😛 )
- ✅ Big refactor/improvement on buttons
- ✅ Big refactor/improvement on colors used for sidebar and channel header
- ✅ Marketplace Apps (list and individual info)
- ✅ Call modal (phone icon in channel)


Related : #189 - Thanks for reporting these issues 👌 

Don't hesitate to leave your thoughts 😉

Mattsky commented:
Just tried on Windows desktop client v3.8.7 and web browser, the toggle button doesn't work for me. (Server version is 5.0.0 just to confirm.)

Mattsky commented:
Could this be related? Pulled from logs:

`Jul 27 20:44:03 XXXX rocketchat[2555171]: {"level":50,"time":"2022-07-27T20:44:03.370Z","pid":2555171,"hostname":"XXXX","name":"rocketchat:theme","err":{"type":"LessError","message":"Unrecognised input","filename":"input","index":5109,"line":184,"callLine":null,"column":42,"extract":["\t/* Tags */","\t--rcx-tag-colors-default-background-color: var(--color-dark);","}"]},"msg":"Unrecognised input"}`

Mattsky commented:
Yup, missing semicolon on line 181. Seems to have fixed it!

TBG-FR commented:
> Yup, missing semicolon on line 181. Seems to have fixed it!

Good catch !

KeysBytes commented:
the ``Admin`` label is also dark now :P

TBG-FR commented:
> the `Admin` label is also dark now :P

Looks like I modified a value (`--rcx-tag-colors-default-background-color:`) that also affect roles, you're right ! I fixed it, commit is on its way 😉

ankar84 commented:
Checked all what I found broken before and now all that is fixed by this great PR!
Thank you @TBG-FR for your fantastic job!

aadilayub commented:
Spotted a couple bugs:

- Link previews are dark on hover

![image](https://user-images.githubusercontent.com/31581044/181546945-0bf14e76-13ca-48b1-bf9b-cd24bd3e9cde.png)


- Emoji reaction boxes are white for the current user (other users reaction emojis have the appropriate dark box)

![image](https://user-images.githubusercontent.com/31581044/181548258-4b391ce8-d664-41e0-b5c1-1a095696bb60.png)

- Admin badges are white

![image](https://user-images.githubusercontent.com/31581044/181549796-418e4519-f614-485a-a062-e422aa1137f4.png)

TBG-FR commented:
> Spotted a couple bugs:
> 
> * Link previews are dark on hover
> 
> ![image](https://user-images.githubusercontent.com/31581044/181546945-0bf14e76-13ca-48b1-bf9b-cd24bd3e9cde.png)
> 
> * Emoji reaction boxes are white for the current user (other users reaction emojis have the appropriate dark box)
> 
> ![image](https://user-images.githubusercontent.com/31581044/181548258-4b391ce8-d664-41e0-b5c1-1a095696bb60.png)
> 
> * Admin badges are white
> 
> ![image](https://user-images.githubusercontent.com/31581044/181549796-418e4519-f614-485a-a062-e422aa1137f4.png)

I'll check and fix links and emojis !

About the role badges, I tried to modify CSS variables (colors) instead of elements one by one, so that's the expected look (based on RocketChat official design), is that an issue ? I found it readable 😅

aadilayub commented:
it's definitely readable but imo stands out a bit too much in dark mode. In light mode the role badges are a darker version of the app background color, so maybe in dark mode they should be a lighter version of the background color?

StefanIGit commented:
Works for me, had to restart Linux Client 3.8.7, reload browser GC 103 with server 5.0.0.
Nice Job :+1: :

TBG-FR commented:
> 

I agree, I'll see what I can do

TBG-FR commented:
> it's definitely readable but imo stands out a bit too much in dark mode. In light mode the role badges are a darker version of the app background color, so maybe in dark mode they should be a lighter version of the background color?

I agree, I'll see what I can do

ae5960e8-a6fc-491f-b252-898ecf59af95 commented:
Great work. I'm getting into CSS, so I'll be able to chip in myself sooner or later. Some issues I noticed:
- Hovering over a quote message (which starts with ">") is pure white, message unreadable.
- "Ok" button in the "Read by" menu of a message is incorrectly colored.
- Image download button is blue. Is this intentional? I think it would look nicer in gray.
- Borders of images are pure white, they could be done away with entirely.
- Quote messages have a pure white line on their right when gray would be more suitable.
- In the admin panel "Settings" tab, the gray box containing the word "Settings" and search box should probably be set to the same color as the background instead of light gray.
- The dark them toggle button doesn't seem to be centred with the other icons, home, search, display, etc...
- In the "Directory" area, hovering over the tabs makes the tab text too dark to read. Also, the background of the tab box changes when it would be better if it stayed the same color as it is already.
- In the "Hide" menu when you choose to hide a chat, the "Cancel" button is wrongly colored.
- Should the message box be lighter than the background? I think it would look nicer as the same color as the background.
- Hovering over an emoji in the emoji picker highlights in pure white when it should be darker.
- Emoji skin tone switcher is wrongly colored.
- Editing a message highlights the message in blue which is too bright.

TBG-FR commented:
> ✅ Link previews are dark on hover
> ✅ Emoji reaction boxes are white for the current user (other users reaction emojis have the appropriate dark box)
> ✅ Admin badges are white

TBG-FR commented:
> * Hovering over a quote message (which starts with ">") is pure white, message unreadable.

I do not reproduce that 🤔 (_maybe my recent modifications fixed it ?_)

> * "Ok" button in the "Read by" menu of a message is incorrectly colored.

This button is missing the `.rcx-button--primary` class sadly

> * Image download button is blue. Is this intentional? I think it would look nicer in gray.

It is black for me (_maybe my recent modifications fixed it ?_)

> * Borders of images are pure white, they could be done away with entirely.

Fixed it ! ✅

> * Quote messages have a pure white line on their right when gray would be more suitable.

Fixed it ! ✅

> * In the admin panel "Settings" tab, the gray box containing the word "Settings" and search box should probably be set to the same color as the background instead of light gray.

Fixed it ! ✅

> * The dark them toggle button doesn't seem to be centred with the other icons, home, search, display, etc...

I noticed it too, I'll try to find a fix 🚧 

> * In the "Directory" area, hovering over the tabs makes the tab text too dark to read. Also, the background of the tab box changes when it would be better if it stayed the same color as it is already.

Fixed it ! ✅

> * In the "Hide" menu when you choose to hide a chat, the "Cancel" button is wrongly colored.

Fixed it ! ✅

> * Should the message box be lighter than the background? I think it would look nicer as the same color as the background.

What do you mean by `message box` ?

> * Hovering over an emoji in the emoji picker highlights in pure white when it should be darker.

Fixed it ! ✅
However... 🚧 I can't find the right css rule to modify the color of the tooltip (when hovering an emoji)

> * Emoji skin tone switcher is wrongly colored.

I'll check this one later  🚧 Feel free to do it if you want !

> * Editing a message highlights the message in blue which is too bright.

Fixed it ! ✅

TBG-FR commented:
![image](https://user-images.githubusercontent.com/18146038/181864639-f6f8ce05-4152-4d3d-8301-6b66d330bf3d.png)

ae5960e8-a6fc-491f-b252-898ecf59af95 commented:
Nice

> Hovering over a quote message (which starts with ">") is pure white, message unreadable. **Fixed**

> Image download button is blue. Is this intentional? I think it would look nicer in gray. **Fixed**

> Borders of images are pure white, they could be done away with entirely. **Fixed**

> Plus the others you confirmed fixed are fixed

How do you find it editing the RC CSS? It's really complicated and bloated to me...

TBG-FR commented:
> How do you find it editing the RC CSS? It's really complicated and bloated to me...

I usually use Chrome/Firefox dev tools (F12) on RocketChat pages, that allows me to find and edit elements (and their rules/names) easily. Once I get a rule working, I put it in the css file, copy the whole css file in `Admin` > `Layout` and refresh the page(s) to check if everything is working well.

TBG-FR commented:
@pbaity This PR is ready to merge, we'll be able to make another one for other small things that could be missing, I think 😉

pbaity commented on dark-mode.css:
```diff
@@ -505,32 +575,63 @@ body.dark-mode .rcx-options .rcx-option--selected:hover {
-/* Square (icon) button style */
-body.dark-mode .main-content .rcx-button--square:not(.rcx-button--ghost), /* Default */
-body.dark-mode .main-content .rcx-button--square:is(.rcx-button--ghost, .rcx-button):hover {  /* Hovered or selected */
+/* Same style, for unusual buttons/icons/etc */
```
Not sure how I feel about these comments yet. I understand their use though, so let's try them out for a bit and see what everyone thinks. My only concern is that they may become unwieldy.

TBG-FR replied:
If it was only up to me, I would almost force them, to have a comment/explanation on each rule, that makes it easier to modify the file, but you're right, that can be heavy too !

pbaity replied:
I definitely agree that explanation/comment would be good. The longer I've worked with the CSS, especially as RocketChat versions have changed, the more I want to think of a way to better organize the CSS code. 

Originally the idea was to group the rules by the location of the elements they styled, but so many various elements share classes that it doesn't completely solve the problem.

I'm very open to ideas though - what do you think is a good method of using comments to document the rules? 

Maybe something like:
```
/* main chat window - button - 5.0.0 */
body.dark-mode .main-content .rcx-button--square:not(.rcx-button--ghost), {
```

On that particular example, someone semi-familiar with Rocket.Chat's CSS can deduce what that rule does just by the class names. But it'll be more helpful on the rules with multiple generated class names.

Curious for your thoughts.

pbaity replied:
By the way, I'm not thinking of the format - your tabular format is also fine - I'm asking more about what content you think would be best.

TBG-FR replied:
I totally agree with your comment, and also with the fact that it became messy with RocketChat evolutions and our tries to organize things by sections

I like the idea, however it can quickly become heavy too if we put the location ("main chat window") in every comment 😅 I think I'd keep some "sections" to separate panels, windows, etc. but less than today
 Version number for each rule is a good idea to track things ! 

On that PR I tried to use CSS variables a lot, to be able to remove (or avoid adding) new rules that we would have to change/sort every time, that's also a way to go, maybe we can reduce our number of rules

pbaity commented on dark-mode.css:
```diff
@@ -505,32 +575,63 @@ body.dark-mode .rcx-options .rcx-option--selected:hover {
 
-body.dark-mode .main-content .rcx-button--square:is(.rcx-button--ghost, .rcx-button):focus {
-	background-color: var(--color-darkest);
+/* Inverted style, for buttons on dark backgrounds */
```
Same as above. If this documentation is helpful but ends up needing to be moved, we may be able to move it to a separate document, potentially CONTRIBUTING.md or a new doc.

TBG-FR replied:
I find them useful to find what a CSS rule is associated to, especially when classes change over versions, and I'm afraid that moving them to a separate file would complicate the job

pbaity reviewed (approved):
Thanks for all your hard work, it's looking great! I appreciate all the little fixes too in typos and such. If we can, let's not forget to follow up on any `TODO`s and commented-out rules that we may be able to remove. Once I update my main instance to 5.0.0 I can help with that.

TBG-FR commented:
> Thanks for all your hard work, it's looking great! I appreciate all the little fixes too in typos and such. If we can, let's not forget to follow up on any `TODO`s and commented-out rules that we may be able to remove. Once I update my main instance to 5.0.0 I can help with that.

I agree, we can clean some rules, comments and TODOs 🤞
